File: day02/Day2.java

import java.util.*;
import java.io.*;
public class Day2 {
   public static void main(String[] args) throws FileNotFoundException{
      Scanner input = new Scanner(new File("input.txt"));
      //a(input);
      b(input);
   }

   public static void a(Scanner input) {
      int checksum = 0;
      while(input.hasNextLine()) {
         String line = input.nextLine();
         int min = Integer.MAX_VALUE;
         int max = Integer.MIN_VALUE;
         Scanner i2 = new Scanner(line);
         while(i2.hasNextInt()) {
            int curr = i2.nextInt();
            if(curr < min) {
               min = curr;
            }
            if(curr > max) {
               max = curr;
            }
         }
         checksum += max - min;
      }
      System.out.println(checksum);
   }

   public static void b(Scanner input) {
      int checksum = 0;
      while(input.hasNextLine()) {
         String line = input.nextLine();
         Scanner i2 = new Scanner(line);
         List<Integer> list = new LinkedList<Integer>();
         list.add(i2.nextInt());
         while(i2.hasNextInt()) {
            int curr = i2.nextInt();
            for(int i: list) {
               if(i % curr == 0) {
                  checksum += i/curr;
               } else if (curr % i == 0) {
                  checksum += curr/i;
               }
            }
            list.add(curr);
         }
      }
      System.out.println(checksum);
   }
}

File: day02/Day2.cpp
#include <stdio.h>
using namespace std;

int main () {
   
   FILE * pFile;
   int curr;
   pFile = fopen("input.txt", "r");

   int sheet[16][16];

   for(int i = 0; i < 16; i++) {
      for(int j = 0; j < 16; j++) {
         fscanf(pFile, "%i", &sheet[i][j]);
      }
   }

   rewind(pFile);
   int checksum = 0;
   for(int i = 0; i < 16; i++) {
      int max = 0;
      int min = 10000;
      for(int j = 0; j < 16; j++) {
         if(sheet[i][j] > max) max = sheet[i][j];
         if(sheet[i][j] < min) min = sheet[i][j];
      }
      checksum += max - min;
   }

   printf("Part A Checksum: %d\n", checksum);
   
   checksum = 0;

   for(int i = 0; i < 16; i++) {
      bool found = false;
      for(int j = 0; j < 16; j++) {
         int c = sheet[i][j];
         for(int k = 0; k < 16; k++) {
            int c2 = sheet[i][k];
            if(k != j) {
               if(c % c2 == 0) {
                  checksum += c / c2;
                  found = true;
                  break;
               } else if(c2 % c == 0) {
                  checksum += c2 / c;
                  found = true;
                  break;
               }
            }
         }
         if(found) break;
      }
   }

   printf("Part B Checksum: %d\n", checksum);

   return 0;
}